exasol_integration_test_docker_environment/lib/base/luigi_log_config.py

```diff
@@ -37,43 +41,51 @@ def get_log_path(job_id: str) -> Path:
+    if restore_logger:
+        before_logger = logger_creator()
+        logger_info = {
+            LOG_LEVEL: before_logger.level,
+            HANDLERS: list(before_logger.handlers),
+            FILTERS: list(before_logger.filters),
+            PROPAGATE: before_logger.propagate
         }
     yield
-    if use_job_specific_log_file and root_logger_info is not None:
-        logging.root.level = root_logger_info[LOG_LEVEL]
-        logging.root.handlers = root_logger_info[HANDLERS]
-        logging.root.filters = root_logger_info[FILTERS]
-        logging.root.propagate = root_logger_info[PROPAGATE]
+    if restore_logger and logger_info is not None:
+        after_logger = logger_creator()
+        after_logger.level = logger_info[LOG_LEVEL]
+        after_logger.handlers = logger_info[HANDLERS]
+        after_logger.filters = logger_info[FILTERS]
+        after_logger.propagate = logger_info[PROPAGATE]
 
 
 @contextlib.contextmanager
 def get_luigi_log_config(log_file_target: Path,
                          use_job_specific_log_file: bool,
-                         console_log_level: Optional[str] = None) -> Path:
+                         log_level: Optional[str] = None) -> Path:
     """
     Yields a context manager containing the path of the log-config file.
     log_file_target contains the location of the log-file.
     console_log_level indicates the log_level (@see https://docs.python.org/3/library/logging.html#logging-levels)
     The log-level for the log-file is always logging.DEBUG!
     """
-    if console_log_level is None:
-        console_log_level = logging.getLevelName(logging.WARNING)
+    if log_level is None and use_job_specific_log_file:
+        log_level = logging.getLevelName(logging.WARNING)
     env = jinja2.Environment(loader=jinja2.PackageLoader(PACKAGE_NAME),
                              autoescape=jinja2.select_autoescape())
     template = env.get_template("luigi_log.conf")
-    rendered_template = template.render(console_log_level=console_log_level,
+    rendered_template = template.render(console_log_level=log_level,
                                         log_file_target=str(log_file_target))
     with tempfile.TemporaryDirectory() as temp_dir:
         temp_luigi_conf_path = Path(temp_dir) / "luigi_log.conf"
         with open(temp_luigi_conf_path, "w") as f:
             f.write(rendered_template)
-        with restore_root_logger(use_job_specific_log_file=use_job_specific_log_file):
+        with restore_logger(logger_creator=lambda: logging.root), \
+                restore_logger(logger_creator=lambda: logging.getLogger(LUIGI_INTERFACE_LOGGER)), \
+                restore_logger(logger_creator=lambda: logging.getLogger(LUIGI_LOGGER)):
+            if log_level is not None and not use_job_specific_log_file:
+                logging.getLogger(LUIGI_INTERFACE_LOGGER).level = logging.getLevelName(log_level)
+                logging.getLogger(LUIGI_LOGGER).level = logging.getLevelName(log_level)
```

Comment:
```suggestion
        with ExitStack() as stack:
            stack.enter_context(restore_logger(logger_creator=lambda: logging.root))
            stack.enter_context(restore_logger(logger_creator=lambda: logging.getLogger(LUIGI_INTERFACE_LOGGER)))
            stack.enter_context(restore_logger(logger_creator=lambda: logging.getLogger(LUIGI_LOGGER)))
            if log_level is not None and not use_job_specific_log_file:
                logging.getLogger(LUIGI_INTERFACE_LOGGER).level = logging.getLevelName(log_level)
                logging.getLogger(LUIGI_LOGGER).level = logging.getLevelName(log_level)
```
You could make use of a [Exit Stack](https://docs.python.org/3/library/contextlib.html#contextlib.ExitStack) here.